localization/ndt_scan_matcher/src/ndt_scan_matcher_core.cpp

```diff
@@ -429,14 +429,23 @@ void NDTScanMatcher::callback_sensor_points(
 
   // perform ndt scan matching
   key_value_stdmap_["state"] = "Aligning";
-  const NdtResult ndt_result = align(interpolator.get_current_pose().pose.pose);
+  const Eigen::Matrix4f initial_pose_matrix =
+    pose_to_matrix4f(interpolator.get_current_pose().pose.pose);
+  const pclomp::NdtResult ndt_result = ndt_ptr_->executeScanMatching(initial_pose_matrix);
   key_value_stdmap_["state"] = "Sleeping";
 
   const auto exe_end_time = std::chrono::system_clock::now();
   const double exe_time =
     std::chrono::duration_cast<std::chrono::microseconds>(exe_end_time - exe_start_time).count() /
     1000.0;
 
+  const geometry_msgs::msg::Pose result_pose_msg = matrix4f_to_pose(ndt_result.pose);
+  std::vector<geometry_msgs::msg::Pose> transformation_msg_array;
+  for (auto pose_matrix : ndt_result.transformation_array) {
```

Comment:
I think `for (const auto& pose_matrix : ndt_result.transformation_array)` is better.

Reply:
https://github.com/autowarefoundation/autoware.universe/pull/2199/commits/717c6161d41f4f0c6f123fbe27849a14d01344f9
Resolved